Added Sign agreement probability implementation (for foldednormal) with test cases

huNterrchen opened the pull request.

huNterrchen committed:
Implement SignAgreementProbability for FoldedNormalSample

To specify it only for FoldedNormalSample, I have to put it in this file without changing the order of inclusion

huNterrchen committed:
Added test cases for SignAgreementProbability

huNterrchen committed:
Minor correction to test case

nignatiadis commented on src/samples/foldednormal.jl:
```diff
@@ -255,3 +255,46 @@ function marginalize(Z::FoldedNormalSample, prior::Uniform)
     unif_normal = marginalize(Z_unfolded, prior)
     fold(unif_normal)
 end
+
```
why introduce such asymmetry between nrmal and foldednormal?

huNterrchen committed:
Updated SignAgreementProbability structures and functions

huNterrchen committed:
Updated SignAgreementProbability

huNterrchen committed:
Updated test cases for SignAgreementProbability with normal sample

huNterrchen committed:
Fix test cases in SignAgreementProbability

huNterrchen committed:
Updated SignAgreementProbability with new definitions

Added definitions for P_G and adjusted the handling of Z=0 in SignAgreementProbability.

huNterrchen committed:
Update interval definitions to new definitions

huNterrchen committed:
Update tests to new definitions for sign agreement probability

huNterrchen committed:
Updated test cases for SignAgreementProbability

huNterrchen committed:
Update documentation for SignAgreementProbability

huNterrchen committed:
Moved SignAgreementProbability for normal sample

huNterrchen committed:
Moved sign agreement probability for normal sample

huNterrchen committed:
Merge branch 'master' into SignAgreementProbability
